picasso/src/main/java/com/squareup/picasso/BitmapHunter.java

```diff
@@ -446,19 +446,29 @@ static Bitmap transformResult(Request data, Bitmap result, int exifRotation) {
       if (data.centerCrop) {
         float widthRatio = targetWidth / (float) inWidth;
         float heightRatio = targetHeight / (float) inHeight;
-        float scale;
+        float scaleX, scaleY;
         if (widthRatio > heightRatio) {
-          scale = widthRatio;
+          scaleX = widthRatio;
+          scaleY = widthRatio;
           int newSize = (int) Math.ceil(inHeight * (heightRatio / widthRatio));
           drawY = (inHeight - newSize) / 2;
           drawHeight = newSize;
+          int rounded = Math.round(drawHeight * scaleY);
+          if (rounded != targetHeight) {
```

Comment:
It's probably okay just to set it. I don't have a good reason for doing it this way. I just didn't want to introduce any additional rounding errors.

I'll update it.